File: src/ComposeUI.Messaging.Core/Messages/ConnectResponse.cs
namespace ComposeUI.Messaging.Core.Messages;

public sealed class ConnectResponse : Message
{
    public ConnectResponse(Guid clientId)
    {
        ClientId = clientId;
    }

    public override MessageType Type => MessageType.ConnectResponse;
    public Guid ClientId { get; init; }
}

File: Tryouts/Messaging-JS/src/protocol/messages/ConnectResponse.ts
import { Error } from "../Error";
import { Message } from "./Message";

export interface ConnectResponse extends Message {
    type: "ConnectResponse";
    clientId?: string;
    error?: Error;
}

export function isConnectResponse(message: Message): message is ConnectResponse {
    return message.type == "ConnectResponse";
}